### PR TITLE
Feat/image card

### PR DESCRIPTION
**Pull request by Amber-Song**
# Context

Link to [Notion ticket](https://www.notion.so/cloudcannon/Image-Card-d80293c2435749e49c487fc1538bbe63?pvs=4)

# Additional information

The button and image component that need to be used in this branch is in working at the moment. 
And I didn't set the width for it because I assume that will be set in grid block (parent) component.

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in Bookshop browser
- The site in CloudCannon (link here)

# Before merge (PR owner)

- [ ] Delete the test site in CloudCannon
- [x] For "generic" components **only**: remove `- content_blocks` from `structures` in the YAML file (this was added for testing).

### CODE REVIEW

File: component-library/components/simple/image-card/image-card.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.c-image-card {
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    padding: 1.5rem;
+
+    &--content {
```
Comment (Yoda-Soda) [inline component-library/components/simple/image-card/image-card.scss]:
```suggestion
    &__content {
```

**Comment by AlyshaNolan** [comment]
From looking at the code I have a couple of thoughts:
    - The gap on the heading component has been changed in the css - this needs to become a dynamic (but probably dev only?) option as the gap is 0.5rem for main components but 0.25rem here.
    - The content p element can use the text-block component (but the frontmatter input will stay the same)
    - I also had some thoughts around the image/icon inputs but I believe Jordan is refactoring that part already

**Comment by Amber-Song** [comment]
> - The gap on the heading component has been changed in the css - this needs to become a dynamic (but probably dev only?) option as the gap is 0.5rem for main components but 0.25rem here.

Cool.

> The content p element can use the text-block component (but the frontmatter input will stay the same)

Good point! Done!

>  I also had some thoughts around the image/icon inputs but I believe Jordan is refactoring that part already

I saw that changes. I merged in main branch.

File: component-library/components/simple/image-card/image-card.eleventy.liquid [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<div class={{c}}>
+    {% if is_image %}
+        {% comment %} TODO: put image component here {% endcomment %}
+    {% elsif icon %}
+        {% bookshop "generic/icon" bind:icon %}
+    {% endif %}
+
+    {% if heading %}{% bookshop "generic/component-heading" bind:heading %}{% endif %}
+
+    <div class="{{c}}--content">{% bookshop "generic/text-block" bind:content %}</div>
```
Comment (Yoda-Soda) [inline component-library/components/simple/image-card/image-card.eleventy.liquid]:
```suggestion
    <div class="{{c}}__content">{% bookshop "generic/text-block" bind:content %}</div>
```

**Review by Yoda-Soda — CHANGES_REQUESTED** [review]

**Comment by AlyshaNolan** [comment]
I also have a couple more things to request:
- The image component is ready now so the TODO to add it can be replaced
- The button component is ready now so the TODO to replace the link can be completed
- I wonder if it should just be called card? As it works for image and icon
- The change I requested around the heading gap has not changed
- The change around the text block is great but the input should still be a straight input (not using the subcomponent)
- My thoughts around icon/image was turning it into a structure but maybe we revisit this after the other changes I have mentioned

**Review by Yoda-Soda — APPROVED** [review]

**Comment by Amber-Song** [comment]
> I also have a couple more things to request:
> 
> * The image component is ready now so the TODO to add it can be replaced
> * The button component is ready now so the TODO to replace the link can be completed
> * I wonder if it should just be called card? As it works for image and icon
> * The change I requested around the heading gap has not changed
> * The change around the text block is great but the input should still be a straight input (not using the subcomponent)
> * My thoughts around icon/image was turning it into a structure but maybe we revisit this after the other changes I have mentioned

Done

File: component-library/components/simple/card/card.bookshop.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Metadata about this component, to be used in the CMS
+spec:
+  structures:
+    - content_blocks
+  label: "Card"
+  description: Block that contain image and description
```
Comment (AlyshaNolan) [inline component-library/components/simple/card/card.bookshop.yml]:
```suggestion
  description: Image/icon and a description in a card format
```

File: component-library/components/simple/card/card.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.c-card {
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    padding: 1.5rem;
+
+    &__content {
+        font-family: var(--font-family);
```
Comment (AlyshaNolan) [inline component-library/components/simple/card/card.scss]:
```suggestion
```

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
- Just realised that it doesn’t really make sense to use the heading subcomponent here as it makes the heading hierarchy incorrect. We either need to change the heading component to let the user pick their hierarchy or just hardcode the heading and eyebrow into the card.
- Needs to be able to change background color
- Image needs to have no padding around it (but icon card needs the padding)
- Allow the user to change the link/button text as well as the url

**Review by adon-cloudcannon — COMMENTED** [review]
Changes made should be reviewed now... perhaps by @jtrbuhovic with an eye to how it works in the grid?

**Comment by jtrbuhovic** [comment]
Thanks @adon-cloudcannon! Checking now, I'll make changes in the `feat/add-image-card-grid` branch when fitting these changes in

**Review by jtrbuhovic — APPROVED** [review]
Looks good!

**Review by AlyshaNolan — APPROVED** [review]

**Pull request merged**